packages/ui-components/src/components/Pagination/Pagination.component.js

```diff
@@ -3,117 +3,218 @@
-  pages = null,
+  totalPages = null,
+  pages = undefined,
+  disabled = false,
   isFirstPage = false,
   isLastPage = false,
   onPressPrevious = undefined,
   onPressNext = undefined,
   onSelectChange = undefined,
   onKeyPress = undefined,
+  onBlur = undefined,
+  progress = undefined,
   className = "",
   ...props
 }) => {
+  const [controlPage, setControlCurrentPage] = useState(1)
+  const [controlTotalPage, setControlTotalPage] = useState(1)
+
+  useEffect(() => {
+    setControlCurrentPage(currentPage)
+    // Fallback for the “pages” prop which was used in an earlier version of this component.
+    pages ? setControlTotalPage(pages) : setControlTotalPage(totalPages)
+  }, [currentPage, totalPages, pages])
+
+  const handleInputChange = (event) => {
+    //convert all non-numeric characters to empty string
+    const inputValue = parseInt(event?.target.value)
+    setControlCurrentPage(inputValue)
+  }
+
   const handlePrevClick = (event) => {
+    setControlCurrentPage(controlPage > 1 ? controlPage - 1 : 1)
     onPressPrevious && onPressPrevious(event)
   }
 
   const handleNextClick = (event) => {
+    setControlCurrentPage(controlPage < controlTotalPage ? controlPage + 1 : controlTotalPage)
     onPressNext && onPressNext(event)
   }
 
-  const handleSelectChange = (event) => {
-    onSelectChange && onSelectChange(event)
+  const handleSelectChange = (selected) => {
+    const s = parseInt(selected)
+    setControlCurrentPage(s)
+    onSelectChange && onSelectChange(s)
   }
 
-  const handleKeyPress = (event) => {
+  const handleEnter = (event) => {
     if (event.key === "Enter") {
+      if (controlPage < 1) {
+        setControlCurrentPage(1)
+      } else if (controlPage > controlTotalPage) {
+        setControlCurrentPage(controlTotalPage)
+      }
       onKeyPress && onKeyPress(event)
     }
   }
 
+  const handleBlur = (event) => {
+    if (controlPage < 1) {
+      setControlCurrentPage(1)
+    } else if (controlPage > controlTotalPage) {
+      setControlCurrentPage(controlTotalPage)
+    }
+    onBlur && onBlur(event)
+  }
+
+  const getInputWidthClass = () => {
+    let logLength = isNaN(controlPage) ? 1 : controlPage?.toString().length
+    logLength = logLength > 5 ? 5 : logLength
+    const width = `${(logLength * 0.6 + 2.1).toFixed(1)}rem` // 0.6rem per digit + 2.1rem
+    return { width: width }
+  }
+
   return (
     <div
       className={`juno-pagination juno-pagination-${variant || "default"} ${paginationStyles} ${className}`}
       {...props}
     >
-      <Button icon="chevronLeft" disabled={isFirstPage} onClick={handlePrevClick} title="Previous Page" />
-      {variant ? renderPaginationInnards(variant, currentPage, pages, handleSelectChange, handleKeyPress) : null}
-      <Button icon="chevronRight" disabled={isLastPage} onClick={handleNextClick} title="Next Page" />
+      <Button
+        icon="chevronLeft"
+        disabled={isFirstPage || disabled || progress || controlPage === 1}
```

Comment:
Something in this boolean expression leads to this button to be disabled when rendering the default variant (i.e. only render the prev/next buttons). My guess is it's the `controlPage === 1` because the initial value for `controlPage` is `1`. You probably need another expression in here that ensures button isn't accidentally disabled in the "default" variant.

Reply:
I don't quite understand. Has it perhaps been resolved with the previous comment?